source/MaterialXGenShader/GenContext.cpp

```diff
@@ -3,6 +3,26 @@
 namespace MaterialX
 {
 
+GenContext::GenContext(ShaderGeneratorPtr sg)
+    : _sg(sg)
+{
+    if (!_sg)
+    {
+        throw ExceptionShaderGenError("GenContext must have a valid shader generator");
```

Comment:
Do you foresee any problems with throwing from the constructor? Handling constructor failure by throwing seems to be the recommended solution, if I Google the subject :) But I might be missing something.
We could add a creator method and make the constructor private instead, but then you could never allocate a context on the stack. And a client would need to try/catch the creation non the less.
It's just a safe guard against hard crashing on bad usage, since a valid shader gen pointer is required.